### PR TITLE
Add QEMU/KVM support for Windows 10, Windows 2019

### PR DESCRIPTION
**Pull request by qmfrederik**
This PR adds QEMU/KVM support for Windows 10, Windows 2019 images:

- Adds qemu provisioners. They also mount the `virt-win.iso` image, which contains the virtio drivers & agent
- Injects the virtio drivers (for paravirtualized storage and networking) via `Autounattend.xml`. 
- Adds support for installing the qemu agent in `vm_guest_tools.bat`
- Updates the default Vagrantfile to:
  * Set CPU, memory defaults for libvirt (2 GB/2 CPUs)
  * Enbable Hyper-V enlightment features
  * Disable the default synced folder, unless the vagrant-winrm-syncedfolders plugin is installed, in which case the synced folder will use winrm

### CODE REVIEW

**Comment by qmfrederik** [comment]
@StefanScherer I'm looking into setting up an Azure Pipeline which can build qemu boxes on a regular basis. I've got the windows_10, windows_2019 and windows_2019_docker images working; still working on getting the Windows 7 image complete.

File: answer_files/10/Autounattend.xml [hunk 1]
```diff
@@ -2,6 +2,28 @@
...
+
+            <!--
+                 This makes the VirtIO drivers available to Windows, assuming that
+                 the VirtIO driver disk
+                 (https://alt.fedoraproject.org/pub/alt/virtio-win/latest/images/bin/)
+                 is available as drive E:
+            -->
+            <DriverPaths>
+                <PathAndCredentials wcm:action="add" wcm:keyValue="2">
+                    <Path>E:\viostor\w7\amd64</Path>
```
Comment (StefanScherer) [inline answer_files/10/Autounattend.xml]: Will this make problems for the other packer builders like VMware or Virtualbox if the E: drive isn't there?

File: answer_files/10/Autounattend.xml [hunk 2]
```diff
@@ -2,6 +2,28 @@
...
     <settings pass="windowsPE">
+        <component name="Microsoft-Windows-PnpCustomizationsWinPE"
+            publicKeyToken="31bf3856ad364e35" language="neutral"
+            versionScope="nonSxS" processorArchitecture="amd64"
+            xmlns:wcm="http://schemas.microsoft.com/WMIConfig/2002/State">
+
+            <!--
+                 This makes the VirtIO drivers available to Windows, assuming that
+                 the VirtIO driver disk
+                 (https://alt.fedoraproject.org/pub/alt/virtio-win/latest/images/bin/)
```
Comment (StefanScherer) [inline answer_files/10/Autounattend.xml]: This link seems deprecated as the readme in that folder shows me.

Maybe update it to `https://docs.fedoraproject.org/en-US/quick-docs/creating-windows-virtual-machines-using-virtio-drivers/index.html#virtio-win-direct-downloads` ?

**Comment by StefanScherer** [comment]
@qmfrederik The Azure pipeline sounds interesting. Please share the azure-pipelines.yml if you can.

File: answer_files/10/Autounattend.xml (hunk 2 above)
Comment (qmfrederik) [reply answer_files/10/Autounattend.xml]: Ah, good catch. It looks like I actually used the new repository but kept the link to the old one. Updated.

File: answer_files/10/Autounattend.xml (hunk 1 above)
Comment (qmfrederik) [reply answer_files/10/Autounattend.xml]: I gave that a try, by adding an invalid path - like `K:\Kaboom\w7\amd64`. Setup did not fail, so I guess invalid/non-existent paths are just silently ignored.

**Comment by qmfrederik** [comment]
@StefanScherer Thanks for the review. I updated the PR.

[There's a sample pipeline script in my brancy](https://github.com/qmfrederik/packer-windows/blob/fixes/win-7-pipelines-clean/azure-pipelines.yml), and [this is what the output looks like](https://dev.azure.com/qmfrederik/packer-windows/_build/results?buildId=747&view=logs&j=15507b82-a697-5273-6c71-d3a406fbdc9d). It does rely on an on-premise build agent, though, to get virtualization support.

**Pull request merged**